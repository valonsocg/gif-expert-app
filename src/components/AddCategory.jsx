import PropTypes from "prop-types";
import { useState } from "react";

const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("");

  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };

  const onSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length <= 1) return;
    onNewCategory(inputValue.trim());
    setInputValue("");
  };

  return (
    <form onSubmit={onSubmit} aria-label="form">
      <input
        type="text"
        placeholder="Search GIF"
        onChange={onInputChange}
        value={inputValue}
      />
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
export default AddCategory;
