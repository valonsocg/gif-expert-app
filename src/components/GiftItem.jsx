import PropTypes from "prop-types";
const GiftItem = ({ title, url }) => {
  return (
    <div className="card">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  );
};

export default GiftItem;

GiftItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
