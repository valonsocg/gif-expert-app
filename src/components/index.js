export { default as AddCategory } from "./AddCategory";
export { default as GifGrid } from "./GifGrid";
export { default as GiftItem } from "./GiftItem";
