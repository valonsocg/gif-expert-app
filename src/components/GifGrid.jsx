import GiftItem from "./GiftItem";
import useFetchGifs from "../hooks/useFetchGifs";
import PropTypes from "prop-types";

const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>
      {isLoading && <h2>Loading...</h2>}
      <ol className="card-grid">
        {images.map((image) => (
          <GiftItem key={image.id} {...image} />
        ))}
      </ol>
    </>
  );
};

export default GifGrid;

GifGrid.propTypes = {
  category: PropTypes.func.isRequired,
};
