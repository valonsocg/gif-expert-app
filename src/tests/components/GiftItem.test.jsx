import { render, screen } from "@testing-library/react";
import { test } from "vitest";
import { describe } from "vitest";
import { GiftItem } from "../../components";
import { expect } from "vitest";

describe("testing <GiftItem />", () => {
  const title = "One Punch";
  const url = "https://www/one-punch.com/one.jpg";

  test("debe de hacer match con el snapshot", () => {
    const { container } = render(<GiftItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("debe mostrar la imagen con el URL y el ALT indicado", () => {
    render(<GiftItem title={title} url={url} />);
    // screen.debug();
    // console.log(screen.getByRole("img").alt);
    // expect(screen.getByRole("img").src).toBe(url);
    // expect(screen.getByRole("img").alt).toBe(title);
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test("debe de mostrar el titulo en el componente", () => {
    render(<GiftItem title={title} url={url} />);

    expect(screen.getByText(title)).toBeTruthy();
  });
});
