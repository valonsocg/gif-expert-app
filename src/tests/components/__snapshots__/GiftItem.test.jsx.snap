// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`testing <GiftItem /> > debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One Punch"
      src="https://www/one-punch.com/one.jpg"
    />
    <p>
      One Punch
    </p>
  </div>
</div>
`;
